lib/stripe/resources/account.rb

```diff
@@ -45,12 +45,8 @@ def resource_url
     end
 
     # @override To make id optional
-    def self.retrieve(id = ARGUMENT_NOT_PROVIDED, opts = {})
-      id = if id.equal?(ARGUMENT_NOT_PROVIDED)
-             nil
-           else
-             Util.check_string_argument!(id)
-           end
```

Comment:
This PR is a chunk of work pulled out of #954. For context on this change:
>ARGUMENT_NOT_PROVIDED doesn't play very well with how we merge options. The args object in method_missing is an array and merging options when an ID is omitted results in [nil, {}] being sent to the resource, which isn't desirable.

>However, I wasn't able to re-create any of the scenarios that warranted the addition of it in c269e9b and believe it's safe for removal.

https://github.com/stripe/stripe-ruby/pull/954#discussion_r523850127

Reply:
Thx for checking those. Not sure if it there was something different going on when we originally put this in, but re-reading it again, it looks safe to take it out `ARGUMENT_NOT_PROVIDED.